src/client.ts

```diff
@@ -60,14 +60,13 @@ export default class ClientManager {
     const signer = Wallet.fromMnemonic(mnemonic).privateKey;
     const clientOpts = { signer, store: this._store, ethProviderUrl, nodeUrl, logLevel };
     const client = await connext.connect(network, clientOpts);
-    const initOpts = { network, signer, ethProviderUrl, nodeUrl, logLevel };
-    await this.updateClient(client, initOpts, subscriptions);
```

Comment:
Just to be sure shouldn't we unsubscribe from all subscriptions before re-assigning the client on the ClientManager? This was handled on the updateClient method called on initClient

Reply:
Reassigning the client was causing weird issues so I'm not sure if `updateClient` is now necessary. I tested the subscriptions with xud (it re-subscribes on every restart) and I wasn't getting multiple `CREATED` and `UNLOCKED` events.

With current logic it returns the existing subscription if the subscription for the event already exists. https://github.com/connext/rest-api-client/blob/master/src/subscriber.ts#L24